core/imds/src/main/java/software/amazon/awssdk/imds/Ec2MetadataAsyncClient.java

```diff
@@ -84,5 +90,15 @@ interface Builder extends Ec2MetadataClientBuilder<Ec2MetadataAsyncClient.Builde
          * @return a reference to this builder
          */
         Builder httpClient(SdkAsyncHttpClient httpClient);
+
+        /**
+         * A http client builder used to retrieve an instance of an {@link SdkAsyncHttpClient}. If specified, the Ec2
+         * Metadata Client will use the instance returned by the builder and manage its lifetime by closing the http client
+         * once the Ec2 Client itself is closed.
+         *
+         * @param builder the builder to used to retrieve an instance.
+         * @return a reference to this builder
+         */
+        Builder httpClient(DefaultSdkAsyncHttpClientBuilder builder);
```

Comment:
It should be 

 ```
Builder httpClientBuilder(SdkAsyncHttpClient.Builder builder);
```

https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkSyncClientBuilder.java#L56

https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkAsyncClientBuilder.java#L76